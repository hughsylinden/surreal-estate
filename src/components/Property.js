import React from "react";
import PropTypes from "prop-types";
import "../styles/Property.css";
import bed from "../styles/img/bed.svg";
import bathroom from "../styles/img/bathroom.svg";
import pound from "../styles/img/pound.svg";
import city from "../styles/img/city.svg";
import email from "../styles/img/email.svg";
import type from "../styles/img/house.svg";

const Property = ({ property }) => {
  return (
    <div className="property">
      <div className="property-item title">{property.title}</div>
      <div className="property-item">
        <img className="image" alt="bedrooms" src={bed} />
        {property.bedrooms}
      </div>
      <div className="property-item">
        <img className="image" alt="bathrooms" src={bathroom} />
        {property.bathrooms}
      </div>
      <div className="property-item">
        <img className="pound" alt="price" src={pound} />
        {property.price}
      </div>
      <div className="property-item">
        <img className="image" alt="city" src={city} />
        {property.city}
      </div>
      <div className="property-item">
        <img className="type" alt="type" src={type} />
        {property.type}
      </div>
      <div className="property-item email">
        <img className="email-logo" alt="email" src={email} />
        {property.email}
      </div>
    </div>
  );
};

Property.propTypes = {
  property: PropTypes.shape({
    title: PropTypes.string,
    city: PropTypes.string,
    bedrooms: PropTypes.number,
    bathrooms: PropTypes.number,
    price: PropTypes.number,
    email: PropTypes.string,
    type: PropTypes.string,
  }).isRequired,
};
export default Property;
