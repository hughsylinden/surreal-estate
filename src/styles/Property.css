@import url("https://fonts.googleapis.com/css2?family=Anton&display=swap");

.property {
  display: flex;
  flex-direction: column;
  background-color: white;
  border-radius: 25px;
  width:275px;
  margin-top: 10px;
  margin-right: 20px;
  margin-left: 20px;
  margin-bottom: 10px;
}

.property-item {
  display: flex;
  padding: 5px;
  align-items: center;
  font-size: medium;
}

.bedrooms__img, .bathrooms__img, .city__img{
  height: 20px;
  padding-right: 5px;
}

.price__img, .type__img, .email__img {
  height: 15px;
  padding-right: 5px;
}

.email {
  padding: 10px;
  justify-content: center;
  font-size: large;
}

.title {
  font-family: "Anton", sans-serif;
  font-size: large;
  justify-content: center;
  margin: 20px;
  border: solid 1px #393d3f;
  border-radius: 25px;
  background-color: #f1e8b8;
  color:#393d3f;
}