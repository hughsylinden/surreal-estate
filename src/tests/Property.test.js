import { render, screen } from "@testing-library/react";
import React from "react";
import Property from "../components/Property";

describe("Property", () => {
  const dummyProperty = {
    title: "dummyTitle",
    bedrooms: 1,
    bathrooms: 2,
    price: 3,
    city: "dummyCity",
    type: "dummyType",
    email: "dummyEmail",
  };
  test("renders Property component", () => {
    const { asFragment } = render(<Property property={dummyProperty} />);
    expect(asFragment).toMatchSnapshot();
  });

  test("renders Property component with property props as elements", () => {
    render(<Property property={dummyProperty} />);

    expect(screen.getByText("dummyTitle")).toHaveClass("property-item title");
    expect(screen.getByText("1")).toHaveClass("property-item bedrooms");
    expect(screen.getByText("2")).toHaveClass("property-item bathrooms");
    expect(screen.getByText("3")).toHaveClass("property-item price");
    expect(screen.getByText("dummyCity")).toHaveClass("property-item city");
    expect(screen.getByText("dummyType")).toHaveClass("property-item type");
    expect(screen.getByText("dummyEmail")).toHaveClass("property-item email");
  });
  test("renders Property component with img elements for each propert", () => {
    render(<Property property={dummyProperty} />);

    expect(screen.getByAltText("bedrooms")).toHaveClass("bedrooms__img");
    expect(screen.getByAltText("bathrooms")).toHaveClass("bathrooms__img");
    expect(screen.getByAltText("price")).toHaveClass("price__img");
    expect(screen.getByAltText("city")).toHaveClass("city__img");
    expect(screen.getByAltText("type")).toHaveClass("type__img");
    expect(screen.getByAltText("email")).toHaveClass("email__img");
  });
});
