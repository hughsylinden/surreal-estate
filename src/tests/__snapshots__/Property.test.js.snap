// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property renders Property component 1`] = `[Function]`;
